### PR TITLE
Fix inline docs

### PR DESCRIPTION
**Pull request by odersky**
A PR to work together on the inline docs.

### CODE REVIEW

**Comment by odersky** [comment]
@aggelos I pushed fixes to the rest of inline.md. 

One more thing to do: "erased terms" should go into the meta programming section. We should have a new toc entry for this also.

**Comment by odersky** [comment]
@nicolasstucki Can you also go over macros.md, and the following pages to verify that everything is up to date?

File: docs/docs/reference/metaprogramming/inline.md [hunk 1]
```diff
@@ -246,12 +218,42 @@ inline def zero() <: Int = 0
...
+simplify.
+
+Example:
+
+```scala
+inline def update(delta: Int) =
+  inline if (delta >= 0) increaseBy(delta)
+  else decreaseByf(delta)
+```
+A call `update(22)` would rewrite to `increaseBy(22`. But if `update` was called with
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/inline.md]:
```suggestion
A call `update(22)` would rewrite to `increaseBy(22)`. But if `update` was called with
```

File: docs/docs/reference/metaprogramming/inline.md [hunk 2]
```diff
@@ -246,12 +218,42 @@ inline def zero() <: Int = 0
...
+the selected branch. Prefixing an if-then-else expression with `inline` forces
+the condition to be a constant, and thus guarantees that the conditional will always
+simplify.
+
+Example:
+
+```scala
+inline def update(delta: Int) =
+  inline if (delta >= 0) increaseBy(delta)
+  else decreaseByf(delta)
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/inline.md]:
```suggestion
  else decreaseBy(-delta)
```

File: docs/docs/reference/metaprogramming/inline.md [hunk 3]
```diff
@@ -287,12 +289,11 @@ val intTwo: 2 = natTwo
 
 `natTwo` is inferred to have the singleton type 2.
 
-#### scala.compiletime._
+## The scala.compiletime Package
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/inline.md]:
```suggestion
## The `scala.compiletime` Package
```

**Pull request merged**